### PR TITLE
Add search by property name

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java [hunk 1]
```diff
@@ -1016,7 +1016,8 @@ public ArtifactSearchResultsDto searchArtifacts(Set<SearchFilter> filters, Order
                                 + "v.groupId LIKE ? OR "
                                 + "a.artifactId LIKE ? OR "
                                 + "v.description LIKE ? OR "
-                                + "EXISTS(SELECT l.globalId FROM labels l WHERE l.label = ? AND l.globalId = v.globalId)"
+                                + "EXISTS(SELECT l.globalId FROM labels l WHERE l.label = ? AND l.globalId = v.globalId) OR "
+                                + "EXISTS(SELECT p.globalId FROM properties p WHERE p.pkey = ? AND p.globalId = v.globalId)"
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java:1020]: Should we also include property values here?  As another **OR** clause?

**Comment by EricWittmann** [comment]
This is a good iterative improvement.  However, since properties are key-value pairs, I think it would be interesting to improve the Search API to allow both:

1) Search for artifacts by existence of a property key (what this PR does)
2) Search for artifacts by property name **and** value.  So all artifacts with a property `foo` with value `bar`.

Thoughts?  When I designed the API, this is what I wrote for the description of the `properties` query param:

> Filter by one or more name/value property.  Separate each name/value pair using a colon.  For
> example `properties=foo:bar` will return only artifacts with a custom property named `foo`
> and value `bar`.

So something like this:

`/search/artifacts?offset=0&limit=10&properties=foo:bar&properties=baz`

That would return only artifacts with a property named `baz` (with any value) **and** with a property named `foo` that has a value of `bar`.

What do you think?

File: app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java (hunk 1 above)
Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java:1020]: Yes, I think so.

**Comment by carlesarnal** [comment]
Yes, that's exactly the idea behind this PR. I just implemented this as a first step in an iterative approach. The idea is that subsequent PRs will contain search by both name and value and also UI changes.

**Pull request merged**